backend/de.metas.util/src/main/java/de/metas/util/collections/CollectionUtils.java

```diff
@@ -392,4 +390,10 @@
 				.collect(ImmutableList.toImmutableList());
 	}
 
+	@NonNull
+	public static <T> ArrayList<T> mergeLists(@NonNull final ArrayList<T> list1,@NonNull final ArrayList<T> list2)
+	{
+		list1.addAll(list2);
+		return list1;
+	}
```

Comment:
@fp-ps  
cc @metas-ts 

pls avoid introducing methods which has side-effects.
In this case, the method is changing the first list instead of creating a new list by concatenating those 2 lists.

More, looking where this method is used I saw the use case was to have a convenient remapping function for Map.merge when the value is a List.

Basically, in this case, you could use `ArrayListMultimap` and the mergeLists convenient method won't be needed.

PS: now leave it like it is, but in future pls consider Guava's multimaps.